### PR TITLE
Fast Fetch after swipe back does not send ad request

### PR DESCRIPTION
**Pull request by keithwrightbos**
Steps to reproduce:

- Within AMP viewer go to page that supports and has Fast Fetch enabled
- Swipe to another page.  Creative frame is removed and state is reset.
- Swipe back.  onLayoutMeasure does not execute therefore no ad request is sent

Adds resumeCallback which will call onLayoutMeasure if measure is not already required.

Additionally added the following:

- Remove separate boolean to track if layout measure was executed and instead rely on adPromise being null
- Report if adPromise is null in layoutCallback when not expected
- Fix issue where amp-analytics element is not being removed from adsense/doubleclick extensions as insertAnalyticsElement was not returning the element

cc/ @ampproject/a4a

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -330,6 +327,12 @@ export class AmpA4A extends AMP.BaseElement {
      * {?HTMLIframeElement}
      */
     this.iframe = null;
+
+    /**
+     * @return whether most recent ad request was generated as part of resume
```
Comment (dvoytenko) [inline extensions/amp-a4a/0.1/amp-a4a.js]: You mean `@type {???}`. Can you also annotate it with `TODO` for future cleanup?

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -420,25 +423,38 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /** @override */
-  onLayoutMeasure() {
-    if (this.xOriginIframeHandler_) {
-      this.xOriginIframeHandler_.onLayoutMeasure();
+  resumeCallback() {
+    this.protectedEmitLifecycleEvent_('resumeCallback');
```
Comment (dvoytenko) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Do you need to add it to the `LIFECYCLE_STAGES` above?

**Comment by dvoytenko** [comment]
LGTM w/couple notes

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 1 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Done

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 2 above)
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Done good catch

**Pull request merged**